File: frontend/src/app/models/workout-history-preview.ts
import {getDuration} from "./workout";

export class WorkoutHistoryPreview {
  id: number;
  title: string;
  startTime: Date;
  endTime: Date;
  totalWeight: number | null;
  prReps: number;

  constructor(data: Partial<WorkoutHistoryPreview>) {
    this.id = data.id || 0;
    this.title = data.title || '';
    this.startTime = data.startTime ? new Date(data.startTime) : new Date();
    this.endTime = data.endTime ? new Date(data.endTime) : new Date();
    this.totalWeight = data.totalWeight !== undefined ? data.totalWeight : null;
    this.prReps = data.prReps || 0;
  }

  public getWorkoutDuration(): number {
    return getDuration(this.startTime, this.endTime);
  }
}

File: backend/src/main/kotlin/com/stevi/workoutbuddy/domain/workout/model/response/WorkoutHistoryPreview.kt
package com.stevi.workoutbuddy.domain.workout.model.response

import java.time.LocalDateTime

data class WorkoutHistoryPreview(
    val id: Long,
    val title: String,
    val startTime: LocalDateTime,
    val endTime: LocalDateTime,
    val totalWeight: Double?,
    val prReps: Long,
)